### ISSUE TITLE
Avoid Crossing Perimeters: Not very effective?

### ISSUE THREAD

**Issue by CCS86** [alexrj/Slic3r#3019]
Hi, I am running Slic3r 1.2.9 and am not seeing  a very effective result from the avoid crossing perimeters function. I have a print that could pretty easily be done without ever crossing a perimeter, yet a have many, many moves that do so:

![image](https://cloud.githubusercontent.com/assets/12439543/8892653/3833f7c0-332b-11e5-8cc6-3b320d794c39.png)

Without that setting enabled, you can see an increase in these moves, but it isn't very dramatic:

![image](https://cloud.githubusercontent.com/assets/12439543/8892654/859256d8-332b-11e5-978d-f7e70b5a00e9.png)

It seems like there is room to optimize that algorithm.

Thanks!

**Comment by SharkAttackOmNom** [comment]
Try out various Seam Positions, between Aligned, Random, and Closest. (closest would probably provide the desired result.

**Comment by CCS86** [comment]
Thanks for the suggestion. I did try that, but saw very little change.

**Comment by CCS86** [comment]
Here it is set to nearest: 

![image](https://cloud.githubusercontent.com/assets/12439543/8896416/92a483fa-33c1-11e5-908b-d004cbef5c5f.png)

**Comment by SharkAttackOmNom** [comment]
While I understand the desire for improved code, lets try out some other work-arounds. It looks like these are all because of the infill patern and Slic3r probably jumping from optimal Infill paths. Doing Rectilinear paterns through a circular object can be a bit..cumbersome. 

Try out other infills and even numbers of shells. Setting the shells to a ridiculous number (like 1000) might cure the issue entirely.

**Comment by CCS86** [comment]
I am happy to test other infill patterns; what do you recommend? 

I want a sparse infill, so fudging the number of shells isn't an option.

Rectilinear way not be ideal on a circular object, but it should still work. Cura can slice with rectilinear infill and has very few moves through the interior:

![image](https://cloud.githubusercontent.com/assets/12439543/8897000/37393694-33d2-11e5-97f1-30d8131ccf98.png)

**Comment by gege2b** [comment]
maybe try with the "concentric" infill pattern, playing with the density to have the sparse infill you look at

**Comment by CCS86** [comment]
Concentric infill just isn't going to give me the result that I want. I need to have some strength.

**Comment by c-born** [comment]
As I recall the last version where Avoid Crossing Perimeters actually worked properly was 0.9.10b
You could give that a try, although you lose a lot of new features.
I'd be interested in seeing the result

**Comment by CCS86** [comment]
It's good to know that it was previously working correctly.

I hope it is acknowledged as a bug for future repair.

**Comment by krizzap** [comment]
I agree 1000% this needs to be fixed, its beyond irritating that this feature doesn't hardly work, and fudging other things isn't and shouldn't be the answer. Not to go there but Cura does a good job of this as stated above, why can't Slic3r?

**Comment by CCS86** [comment]
Hmm, Slic3r dev is a little sleepy these days.

**Comment by lordofhyphens** [comment]
On 08/17/2015 06:26 PM, CCS86 wrote:
>
> Hmm, Slic3r dev is a little sleepy these days.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/alexrj/Slic3r/issues/3019#issuecomment-131991103>.
>
It comes and goes.

-- 
--Joseph Lenox, BS, MS
I'm an engineer. I solve problems.

**Comment by RaabenFuchs** [comment]
Hello everyone, i'm joining this issue, because i print a power supply case at the moment with vertical ventilation holes-> many islands

this causes ridiculous travel moves (every island one time across the whole surface, half arround the whole object, and then a y axis move across the wall) arround the whole object!
(win 64 v1.3.0)

hope you can estimate the moves a bit, really funny:

![InkedIMG_20190309_2154148](https://user-images.githubusercontent.com/36605089/54077393-d5819b00-42b7-11e9-8c41-cd85832d7068.jpg)


i'm total newby, but really prefer  slicer over cura! thx a lot
